### DISCUSSION TITLE
Data for response rate by day

### DISCUSSION THREAD

**Discussion by LeaVerou** [Devographics/surveys#189] [General Survey Design]
Having data for submitted responses by day each survey was open, would help a lot in planning the [roadmap](https://github.com/Devographics/surveys/discussions/112). Namely, we're trying to answer the question: "how long after launch will we start having enough responses to get useful data?". This is important for Interop planning.

We'd need graphs for both responses per day, and cumulative total responses per day, perhaps by grouping `finishedAt` or `updatedAt` by day.
Cumulative responses is more important, but responses per day will help us see the rate and how it drops off over time.
@shainerosewel is this something you may be able to calculate?

**Comment by SachaG** [comment]
We can use the earliest `createdAt` date as the `t=0` survey start, removing any outliers due to testing before the survey opens officially.

**Comment by ShaineRosewel** [comment]
Not sure if this is possible. I would have to check. 

Hi @SachaG, can you once again, provide the definition of the following columns here (so it is easy to find). Thanks!


`finishedAt`
`updatedAt`
`createdAt`

**Reply by SachaG** [reply]
- `createdAt`: when the response document was first created
- `updatedAt `: the last time the document was saved
- `finishedAt `: the last time the user clicked the "finish survey" button on the last page

**Comment by ShaineRosewel** [comment]
Not sure about the initial spike on the chart. I'll check with the other 2 surveys, too. But what i did is to just count the number of unique respondents for each of `createdAt` and `updatedAt`. We do not have `finishedAt` in CSS 2023 (Is this really the case, @SachaG ). Other available date data in `generatedAt`

@SachaG, may I know when State of CSS 2023 was officially launched?

<img width="1149" alt="Screenshot 2023-08-17 at 4 22 40 PM" src="https://github.com/Devographics/surveys/assets/62248526/5fb7179c-a9d9-4560-89af-12a6819fc654">

Please let me know if this is what we need. Assuming that we do not have problem with the columns that we are using, then it is sensible to say we have enough data after the first two weeks.

**Reply by ShaineRosewel** [reply]
Something I'm thinking about too, is that the plot behavior for both `updatedAt` and `createdAt` are almost the same. Based from the column definition, is this expected @SachaG?

**Reply by SachaG** [reply]
It was officially launched on June 15 so the charts make sense. The second big spike is when Kevin Powell made a video about the survey, I think. 

And yeah `createdAt` and `updatedAt` should be very similar since people usually won't take more than a day to finish the survey.

**Reply by LeaVerou** [reply]
Thank you so much @ShaineRosewel, these are great!!

**Comment by ShaineRosewel** [comment]
This one is for CSS 2022. Cumulative sum steepest on the first week.

<img width="1056" alt="Screenshot 2023-08-17 at 4 44 28 PM" src="https://github.com/Devographics/surveys/assets/62248526/5890b73d-b1d2-4983-ba72-665aea511fd4">

**Comment by ShaineRosewel** [comment]
This last one is for JS. We got lots of responses til the third week. I am thinking there must be other factors, too. Not sure about how the survey is being advertised but if after the first say two weeks there is an influential person who advertised the survey, it could contribute to the unexpected peak during the third week. Please let me know 

<img width="934" alt="Screenshot 2023-08-17 at 4 58 08 PM" src="https://github.com/Devographics/surveys/assets/62248526/ee5181b2-47d7-4e4a-a43b-bd0fafbff51c">

**Comment by ShaineRosewel** [comment]
For these, we can say that we get majority of the respondents within the first two weeks of launch, assuming we did all the essential advertising stuff just on time prior or on the launch date. On the fourth week, we get the least addition to responses.

**Reply by LeaVerou** [reply]
This is fantastic, thank you so much @ShaineRosewel. I can already use this, but if you could easily calculate how long it took for each survey to get 25%, 50%, 75% of responses, that would be useful too.

**Reply by ShaineRosewel** [reply]
Thanks, @LeaVerou. 

To address this,

> how long it took for each survey to get 25%, 50%, 75% of responses

I only used `updatedAt` date since it's  always almost equal to `createdAt` date values. Note that the size of the points correspond to the number of respondents (the larger the point, the higher the number of respondents)

Below shows that CSS 2023 took the least duration (less than ten days) to complete 75% of its total, but size of circles also shows it has the least total number of respondents. Also, the difference between its 50% and 75% is steep which meant that some event or announcement or ad must have triggered that boost. CSS 2022 and JS 2022 are more comparable since they both took a longer period of time to complete their 75%. Among the three, JS 2022 has the most number of respondents.

<img width="855" alt="Screenshot 2023-08-20 at 12 14 12 AM" src="https://github.com/Devographics/surveys/assets/62248526/60d3c0fe-c155-4905-80f4-d3b1669b3322">

**Reply by LeaVerou** [reply]
Thank you so much, this is great!

**Reply by LeaVerou** [reply]
I just realized this only has the 3 quartiles as data. I was wondering if we could have a more granular table/graph though, with points for each day post-launch (e.g. day 1, 5%, day 2, 7% etc). Table is fine if quicker to generate and might actually be better (though it would be good to share an actual table/google sheets/coda doc/CSV so I can import it into tooling and do my own calculations). I'm trying to find what the point would be to maximize % of responses while minimizing days post-launch, so we can pick a good point to collect preliminary data to feed it into Interop 2024.

Using `updatedAt` seems fine. It would be even better to use `finishedAt` where available, and `updatedAt` for unfinished responses (though if it's too much work to change that now, feel free to continue using `updatedAt` only). Feel free to scope it down to the two State of CSS surveys, since I think the audience of State of HTML is more similar to that than State of JS (though if including State of JS is no more work, it's nice to have it).

If this is too much work (> 40min), feel free to skip it. It's a nice to have, we already have plenty of data to make informed decisions, thanks to your efforts!

**Reply by ShaineRosewel** [reply]
Sending CSV in a while!

<img width="996" alt="Screenshot 2023-08-23 at 4 45 13 PM" src="https://github.com/Devographics/surveys/assets/62248526/fdd5f246-9cd9-4aa2-acd7-a8b9aeebd10a">

**Reply by ShaineRosewel** [reply]
Let me know if there is any issue. :)
[daily_response.csv](https://github.com/Devographics/surveys/files/12417225/daily_response.csv)

**Reply by LeaVerou** [reply]
Thanks! I created some interactive graphs here: https://coda.io/@leaverou/state-of-html/response-rate-20

**Reply by ShaineRosewel** [reply]
Awesome. Please let me know if there is anything else i can work on.

**Comment by ShaineRosewel** [comment]
Here are the exact values, should you need them. Let me know if this works. :D

<img width="1242" alt="Screenshot 2023-08-20 at 12 28 42 AM" src="https://github.com/Devographics/surveys/assets/62248526/4e6ac7d2-a842-4338-83b9-c6e9ed190bc1">
